File: Semestru 4/MPP/lab/java/lab3/Services/src/main/java/org/example/services/ServiceChild.java
package org.example.services;

import org.example.domain.Child;
import org.example.repository.interf.IChildRepository;

public class ServiceChild implements IServiceChild{
    private IChildRepository Repo;

    public ServiceChild(IChildRepository repo) {
        this.Repo = repo;
    }

    public Iterable<Child> GetAll() {
        return this.Repo.FindAll();
    }

    public Child GetById(int id) {
        return this.Repo.FindById(id);
    }

    public Child AddChild (String name, String cnp) {
        return this.Repo.Save(new Child(name, cnp));
    }

    public Child UpdateChild (Child child) {
        return this.Repo.Update(child);
    }

    public Iterable<Child> GetByAge (int age) {
        return this.Repo.GetAllByAge(age);
    }
}

File: Semestru 4/MPP/lab/.net/lab3/lab3/Server/service/ServiceChild.cs
using Model.app.domain;
using Persistence.app.repo.@interface;
using Services.services;

namespace Server.app.service
{
	public class ServiceChild : IServiceChild
	{
		private IChildRepository Repo;

		public ServiceChild(IChildRepository repo) =>
			this.Repo = repo;

		public IEnumerable<Child> GetAll() =>
			this.Repo.GetAll();

		public Child? GetById(int id) =>
			this.Repo.GetById(id);

		public Child? Update(Child child) =>
			this.Repo.Update(child);

		public IEnumerable<Child> GetByAge(int age) => 
			this.Repo.GetAllByAge(age);

		public Child Add(string name, string cnp) => this.Repo.Create(new Child(name, cnp));

	}
}
